src/state/slices/opportunitiesSlice/resolvers/index.ts

```diff
@@ -0,0 +1,19 @@
+import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+
+import { foxFarmingLpMetadataResolver, foxFarmingLpUserDataResolver } from './foxfarming'
+import type {
+  IDefiProviderToMetadataResolverByDeFiType,
+  IDefiProviderToUserDataResolverByDeFiType,
+} from './types'
+
+export const DefiProviderToMetadataResolverByDeFiType = {
+  [DefiProvider.FoxFarming]: {
+    [DefiType.LiquidityPool]: foxFarmingLpMetadataResolver,
+  },
+} as IDefiProviderToMetadataResolverByDeFiType
+
+export const DefiProviderToUserDataResolverByDeFiType = {
+  [DefiProvider.FoxFarming]: {
+    [DefiType.LiquidityPool]: foxFarmingLpUserDataResolver,
+  },
+} as IDefiProviderToUserDataResolverByDeFiType
```

Comment:
We can tighten these by only casting what we absolutely need to, ensuring we still get type warnings/errors if `IDefiProviderToMetadataResolverByDeFiType` or `IDefiProviderToUserDataResolverByDeFiType` changes:

```suggestion
export const DefiProviderToMetadataResolverByDeFiType: IDefiProviderToMetadataResolverByDeFiType = {
  [DefiProvider.FoxFarming as DefiProvider]: {
    [DefiType.LiquidityPool]: foxFarmingLpMetadataResolver,
  },
}

export const DefiProviderToUserDataResolverByDeFiType: IDefiProviderToUserDataResolverByDeFiType = {
  [DefiProvider.FoxFarming as DefiProvider]: {
    [DefiType.LiquidityPool]: foxFarmingLpUserDataResolver,
  },
}
```

Reply:
Alternatively (and preferably), we can use string interpolated dynamic keys to avoid casting altogether:

```ts
export const DefiProviderToMetadataResolverByDeFiType: IDefiProviderToMetadataResolverByDeFiType = {
  [`${DefiProvider.FoxFarming}`]: {
    [DefiType.LiquidityPool]: foxFarmingLpMetadataResolver,
  },
}

export const DefiProviderToUserDataResolverByDeFiType: IDefiProviderToUserDataResolverByDeFiType = {
  [`${DefiProvider.FoxFarming}`]: {
    [DefiType.LiquidityPool]: foxFarmingLpUserDataResolver,
  },
}
```

Reply:
Wow, that first syntax is exactly what I was searching for! For some reason, I was under the impression you couldn't do type assertions in "object variable as keys" expressions.

That second one looks even cleaner, TIL! 🐐

Reply:
827f50960d929ebd876e488ab4ca1ced74faf561